### PR TITLE
rbac: add cluster scoped CRD ClusterRbacConfig and deprecate RbacConfig.

### PR DESCRIPTION
**Pull request by yangminzhu**
Some notes for reviewers:
* I changed the `RbacConfig` from cluster scope to namespace scope in `pilot/pkg/model/config.go` otherwise the test will fail due to `duplicate message type: "istio.rbac.v1alpha1.RbacConfig"`. I tested this change by fresh install and upgrading Istio. This is okay as Pilot won't create a CRD if it finds it already exists in the k8s cluster.
* I changed the sample config to use `ClusterRbacConfig`. This won't affect the current tasks on istio.io as the change is on `master` branch and the istio.io is using samples in `release-1.0` branch.
* @rshriram, @costinm: I removed a unit test (`destination rule nil traffic policy`) in `route_test.go` as it's depending on the outdated IstioConfigStore fake object. The `DestinationRule()` method has already been removed in #7635. Please let me know if you don't want to remove the unit test.

For more background, see #8825.

### CODE REVIEW

File: pilot/pkg/model/config.go [hunk 1]
```diff
@@ -885,6 +903,22 @@ func (store *istioConfigStore) ServiceRoleBindings(namespace string) []Config {
 	return bindings
 }
 
+func (store *istioConfigStore) MeshRbacConfig() *Config {
+	meshRbacConfig, err := store.List(MeshRbacConfig.Type, "")
+	if err != nil {
+		log.Errorf("failed to get meshRbacConfig: %v", err)
+	}
+	if len(meshRbacConfig) > 1 {
+		log.Errorf("found %d meshRbacConfig, expecting only 1.", len(meshRbacConfig))
```
Comment (diemtvu) [inline pilot/pkg/model/config.go]: As you going to ignore all policies with name is not 'default', this log can be 'warn' level, or even drop it completely.

File: galley/pkg/metadata/kube/types.go [hunk 2]
```diff
@@ -523,6 +523,17 @@ func init() {
 		Converter: converter.Get("identity"),
 	})
 
+	b.Add(kube.ResourceSpec{
+		Kind:      "MeshRbacConfig",
```
Comment (ayj) [inline galley/pkg/metadata/kube/types.go]: @ozevren, should we stick with `Mesh` prefix for cluster-scoped resources or adopt k8s convention and use `Cluster`?

File: pilot/pkg/model/config.go [hunk 3]
```diff
@@ -885,6 +903,22 @@ func (store *istioConfigStore) ServiceRoleBindings(namespace string) []Config {
...
+		if rc.Name == DefaultRbacConfigName {
+			return &rc
+		}
+	}
+	return nil
+}
+
 func (store *istioConfigStore) RbacConfig() *Config {
 	rbacConfigs, err := store.List(RbacConfig.Type, "")
 	if err != nil {
```
Comment (liminw) [inline pilot/pkg/model/config.go]: This should no longer be an error log. If MeshRbacConfig is used, we do not expect RbacConfig exist. Maybe change it to an "info" log.  

Also, if users still use RbacConfig, it is better to raise a warning that "RbacConfig is deprecated. Use MeshRbacConfig instead". The best place is in the validation code when users enters an "RbacConfig".

Comment (yangminzhu) [reply pilot/pkg/model/config.go]: Updated. I logged the warning as the validate framework doesn't support to print a warning message to user.
Later we may change it to allow printing a warning message.

File: galley/pkg/metadata/kube/types.go (hunk 2 above)
Comment (ayj) [reply galley/pkg/metadata/kube/types.go]: I had a quick offline chat with @ozevren. I think we want to move towards using `Cluster` prefix for cluster-scoped k8s resources. Mesh-scoped (i.e. multi-cluster) resources ought to be expressed as another layer on top of cluster-scoped resources.

Comment (yangminzhu) [reply galley/pkg/metadata/kube/types.go]: Sure, I'll update the PR to rename it to `ClusterRbacConfig`.
/cc @diemtvu @liminw

**Comment by yangminzhu** [comment]
@diemtvu 
I'm planning to either implement a tool in `istioctl` or put it in a simple bash script to help user migrate, any suggestions for this?

**Comment by yangminzhu** [comment]
@ayj I have changed to `ClusterRbacConfig`, PTAL, thank you.

**Comment by liminw** [comment]
What is the recommended name for Mesh level /cluster scoped object?  I think authentication policy has MeshPolicy. @ayj @diemtvu @geeknoid @ozevren

**Comment by ayj** [comment]
> What is the recommended name for Mesh level /cluster scoped object? I think authentication policy has MeshPolicy. 

We should work towards using `Cluster` for cluster-scoped resources. It's consistent with k8s naming conventions and accurately represents the scope of these resources. Multi-cluster (i.e. mesh) scoped resources is a layer on-top cluster scoped resources.  We'll need to encode this recommendation as part of Istio API governance / style.

**Comment by diemtvu** [comment]
> @diemtvu
> I'm planning to either implement a tool in `istioctl` or create a simple bash script to help user migrate, any suggestions for this?

sgtm

**Comment by yangminzhu** [comment]
@rshriram @ayj @liminw  I have addressed all comments so far, PTAL. Thanks.

**Pull request merged**